### PR TITLE
Rename keys for any hashes inside arrays. Fixes #35

### LINKED ISSUE #35: Formatting keys does not happen for array
Renaming the keys does not happen for an array of results.
See https://github.com/C-S-D/carrot_rpc/blob/master/lib/carrot_rpc/rpc_client.rb#L84

Possible solution would be to iterate through the array and format the keys for each hash
`result.map!({|r| response_key_formatter(r).with_indifferent_access }) if result.is_a? Array`

### PR DESCRIPTION
**Pull request by thewalkingtoast**
Fixes issue https://github.com/C-S-D/carrot_rpc/issues/35.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,20 +39,24 @@
...
+    - [Incompatible Changes](#incompatible-changes-5)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 # Changelog
 All significant changes in the project are documented here.
 
+## v1.0.0
+### Incompatible Changes
+* [#53](https://github.com/C-S-D/carrot_rpc/pull/53) - Rename keys for any hashes inside arrays. Fixes issue [#35](https://github.com/C-S-D/carrot_rpc/issues/35) - [@thewalkingtoast](https://github.com/thewalkingtoast)
```
Comment (shamil614) [inline CHANGELOG.md]: Definitely want to bump the version. But we need to include the other recent PRs in the changelog
https://github.com/C-S-D/carrot_rpc/pull/52
https://github.com/C-S-D/carrot_rpc/pull/50
https://github.com/C-S-D/carrot_rpc/pull/48

File: lib/carrot_rpc/hash_extensions.rb [hunk 2]
```diff
@@ -5,18 +5,20 @@ module CarrotRpc::HashExtensions
...
     # @param [String] replace the text to replace the found text
     # @return [Hash] a new hash
-    def rename_keys(find, replace, new_hash = {})
+    def rename_keys(find, replace, new_hash = {}) # rubocop:disable Metrics/MethodLength
       each do |k, v|
         new_key = k.to_s.gsub(find, replace)
 
-        new_hash[new_key] = if v.is_a? Hash
+        new_hash[new_key] = if v.is_a? Array
+                              v.map { |t| t.rename_keys(find, replace) }
```
Comment (shamil614) [inline lib/carrot_rpc/hash_extensions.rb]: 👏

File: spec/carrot_rpc/rpc_client_spec.rb [hunk 3]
```diff
@@ -213,17 +213,38 @@ def delete_queue
...
+            ]
+          }
+        end
+
         def show(_params)
           {
             "foo-baz" => {
               "biz-baz" => {
-                "super-duper" => "grovy"
+                "super-duper" => "groovy-enchiladas"
```
Comment (shamil614) [inline spec/carrot_rpc/rpc_client_spec.rb]: mmmmm enchiladas

File: CHANGELOG.md (hunk 1 above)
Comment (thewalkingtoast) [inline CHANGELOG.md]: @shamil614 @nward  Updated those in https://github.com/C-S-D/carrot_rpc/pull/54

**Pull request merged**